### PR TITLE
Backout api changes, call to elementSessions instead

### PR DESCRIPTION
**Pull request by wooj-stripe**
## Summary
Reverts API change

## Motivation
Change in direction of the API

## Testing
Manually tested with payment sheet example, but will need to update tests.

### CODE REVIEW

**Review by davidme-stripe — APPROVED** [review]

**Review by davidme-stripe — APPROVED** [review]

File: Stripe/StripeiOSTests/CustomerAdapterTests.swift [hunk 1]
```diff
@@ -126,29 +126,7 @@ class CustomerAdapterTests: APIStubbedTestCase {
...
-        let sut = StripeCustomerAdapter(customerEphemeralKeyProvider: ekm.getEphemeralKey,
-                                        setupIntentClientSecretProvider: { return "si_" },
-                                        configuration: configuration, apiClient: apiClient)
-        let pms = try await sut.fetchPaymentMethods()
-
-        XCTAssertEqual(pms.count, 2)
-        XCTAssertEqual(pms[0].stripeId, expectedPaymentMethods_card[0].stripeId)
-        XCTAssertEqual(pms[1].stripeId, expectedPaymentMethods_usbank[0].stripeId)
-        await waitForExpectations(timeout: 2)
+    // TODO
```
Comment (davidme-stripe) [inline Stripe/StripeiOSTests/CustomerAdapterTests.swift:129]: ?

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheet.swift [hunk 2]
```diff
@@ -160,13 +164,14 @@ internal enum InternalCustomerSheetResult {
...
 extension CustomerSheet {
-    func loadPaymentMethodInfo(completion: @escaping (Result<([STPPaymentMethod], CustomerPaymentOption?), Error>) -> Void) {
+    func loadPaymentMethodInfo(completion: @escaping (Result<([STPPaymentMethod], CustomerPaymentOption?, STPElementsSession), Error>) -> Void) {
         Task {
             do {
                 async let paymentMethodsResult = try customerAdapter.fetchPaymentMethods()
                 async let selectedPaymentMethodResult = try self.customerAdapter.fetchSelectedPaymentOption()
-                let (paymentMethods, selectedPaymentMethod) = try await (paymentMethodsResult, selectedPaymentMethodResult)
-                completion(.success((paymentMethods, selectedPaymentMethod)))
+                async let elementSessionResult = try configuration.apiClient.retrieveElementsSessionForCustomerSheet()
```
Comment (davidme-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheet.swift:172]: Will this run when the customerAdapter's `canCreateSetupIntents` is false? I can't tell from this func

File: Stripe/StripeiOSTests/CustomerAdapterTests.swift (hunk 1 above)
Comment (wooj-stripe) [reply Stripe/StripeiOSTests/CustomerAdapterTests.swift:129]: I feel like the code is going to change quite a bit when we go to enable usbankaccount,  so I just decided to nuke the test and re-write it when we go to enable usbankaccount.

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheet.swift (hunk 2 above)
Comment (wooj-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheet.swift:172]: It does.. oof, sorry, i missed that.  need to fix this. thank you.

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheet.swift [hunk 3]
```diff
@@ -160,18 +164,27 @@ internal enum InternalCustomerSheetResult {
...
 extension CustomerSheet {
-    func loadPaymentMethodInfo(completion: @escaping (Result<([STPPaymentMethod], CustomerPaymentOption?), Error>) -> Void) {
+    func loadPaymentMethodInfo(completion: @escaping (Result<([STPPaymentMethod], CustomerPaymentOption?, [STPPaymentMethodType]), Error>) -> Void) {
         Task {
             do {
                 async let paymentMethodsResult = try customerAdapter.fetchPaymentMethods()
                 async let selectedPaymentMethodResult = try self.customerAdapter.fetchSelectedPaymentOption()
-                let (paymentMethods, selectedPaymentMethod) = try await (paymentMethodsResult, selectedPaymentMethodResult)
-                completion(.success((paymentMethods, selectedPaymentMethod)))
+                async let supportedPaymentMethodTypes = try self .retrieveSupportedPaymentMethodTypes()
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheet.swift:172]: awkward -- why does this pass lint? lol

**Review by davidme-stripe — APPROVED** [review]

**Pull request merged**